app/services/utm.js

```diff
@@ -41,28 +42,54 @@ export default Service.extend({
     return Object.keys(this.existing).length > 0;
   }),
 
+  searchParams: computed('router.location.path', function () {
+    const search = this.router.location.path.split('?')[1] || '';
+    return new URLSearchParams(search);
+  }),
+
+  hasParamsInUrl: computed('searchParams', function () {
+    return UTM_FIELD_NAMES.any(field => this.searchParams.has(field));
+  }),
+
   peek(fields, includeEmpty = true) {
     return fields.reduce((utmData, field) => {
       const value = this.get(SERVICE_UTM_VARS[field]);
       if (value || includeEmpty) {
         utmData[field] = value;
       }
       return utmData;
-    }, {});
+    }, new QueryParamsHash());
+  },
+
+  capture(forceClear = false) {
+    if (this.hasParamsInUrl) {
+      UTM_FIELD_NAMES.forEach(field => {
+        const value = this.searchParams.get(field);
+        this.set(SERVICE_UTM_VARS[field], value);
+      });
+      if (forceClear) this.clearUrl();
```

Comment:
I don't see a `clearUrl` function, is it supposed to be `removeFromUrl`?

Reply:
Ah, right, thanks for catching 👍 Forgot to rename.